docs/content/developer/tutorials/sponsored-txs.mdx

```diff
@@ -0,0 +1,365 @@
+```
+
+The Gas Station server will have the following `main` function in the `src/main.rs` file:
+
+:::info
+This implementation of a gas station server has a simple implementation of shared state to keep track of addresses in memory. In a real-world application, you would want to use a database to store this information.
+:::
+
+```rust reference
+https://github.com/iota-community/sponsored-transactions-demo/blob/main/backend/src/main.rs#L124-L150
+```
+
+
+We technically have two routes in the Gas Station server:
+1- The root route `/` will return a welcome message.
+2- The `/sign_and_fund_transaction` route will handle the user's request to sponsor the free trial. This route will call the `sign_and_fund_transaction` function, which we will implement next.
+
+:::danger No data persistence
+In a real-world application, you would want to use a database to store the shared state information. However, for simplicity, we are using a shared state in memory. 
+This means that if the server restarts, the shared state will be lost.
+:::
+
+Next, we will implement the `sign_and_fund_transaction` function, which will be called when the user sends a request to the `/sign_and_fund_transaction` route. This function will verify the user's eligibility, build the transaction, sign it, and return the transaction and the sponsor's signature.
+
+
+```rust reference
+https://github.com/iota-community/sponsored-transactions-demo/blob/main/backend/src/main.rs#L63-L122
+```
+
+The function `sign_and_fund_transaction` will:
+1- Check if the user has already requested funds. If they have, it will return a conflict status.
+2- Add the user's address to the set of addresses that have requested funds.
+3- Build a signed and funded transaction using the `sign_and_fund_transaction` util function, which we will implement next.
+4- Increment the total sponsored fees by the gas price. This is used for tracking by the sponsor.
+5- Respond with the transaction bytes and the sponsor's signature.
+
+Next, we will implement the `sign_and_fund_transaction` util function, which will build the transaction, sign it, and return the signed transaction.
+under the `utils.rs` file, add the following code:
+
+```rust reference
+https://github.com/iota-community/sponsored-transactions-demo/blob/main/backend/src/utils.rs#L119-L208
+```
+
+This function will:
+1- Build a programmable transaction that calls the `free_trial` function of the `sponsored_transactions_packages` package.
+2- Get the gas coin of the sponsor.
+3- Create a transaction data object with the programmable transaction, gas budget, gas price, and sponsor address.
+4- Sign the transaction by the sponsor.
+5- Return the signed transaction.
+
+
+Finally, we can now run the server and call the `/sign_and_fund_transaction` route to sponsor the free trial. The server will return the transaction bytes and the sponsor's signature, which the user can use to send the transaction to the IOTA network.
```

Comment:
There's a lot of code in the server repo that hasn't been added to the tutorial, i.e. imports and the faucet related functions.  If the user is simply following this, it won't run, as well; it'd be nice to add links to the refs for all the SDK stuff that you use.

Reply:
I added a note about pasting all code parts into respective files 👍